Crane/app/src/androidTest/java/androidx/compose/samples/crane/details/DetailsActivityTest.kt

```diff
@@ -59,61 +51,67 @@ class DetailsActivityTest {
     var hiltRule = HiltAndroidRule(this)
 
     @get:Rule(order = 1)
-    val composeTestRule = createEmptyComposeRule()
+    val composeTestRule = createComposeRule()
+
+    private lateinit var mapState: CameraPositionState
 
     @Before
     fun setUp() {
         hiltRule.inject()
+
         cityDetails = destinationsRepository.getDestination(MADRID.name)!!
-        scenario = ActivityScenario.launch(
-            createDetailsActivityIntent(
-                InstrumentationRegistry.getInstrumentation().targetContext,
-                testExploreModel
-            )
-        )
-    }
+        mapState = CameraPositionState()
+
+        val countDownLatch = CountDownLatch(1)
+
+        composeTestRule.setContent {
+            CraneTheme {
+                Column {
+                    CityMapView(
+                        latitude = testExploreModel.city.latitude,
+                        longitude = testExploreModel.city.longitude,
+                        cameraState = mapState,
+                        onMapLoaded = {
+                            countDownLatch.countDown()
+                        }
+                    )
+                }
+            }
+        }
 
-    @After
-    fun tearDown() {
-        scenario.close()
+        countDownLatch.await()
     }
 
     @Test
     fun mapView_cameraPositioned() {
-        composeTestRule.onNodeWithText(cityDetails.city.nameToDisplay).assertIsDisplayed()
```

Comment:
Should we keep testing that the name and description of the map is on the screen?

Reply:
Added a new test file for it